packages/react-router/README.md

```diff
@@ -0,0 +1,378 @@
+# `@shopify/react-router`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt) [![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-router.svg)](https://badge.fury.io/js/%40shopify%2Freact-router)
+
+A universal router for React with first-class support for the [WHATWG `URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) and prefetching.
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-router
+```
+
+## Usage
+
+### `<Router />`
+
+Creates a router that is provided to the rest of the React tree. This router powers the `useCurrentUrl()` and `useRouter()` hooks and, by extension, the rest of the features of this library. On the client, you should render a `Router` with no props. On the server, you should pass in the `url` prop, which is a `URL` object that represents the current location to simulation.
+
+```tsx
+// Example in Koa...
+
+const app = new Koa();
+app.use((ctx) => {
+  ctx.body = renderToString(
+    <Router url={ctx.URL}>
+      <App />
+    </Router>,
+  );
+});
+```
+
+This component basically just composes the `Prefetcher` component, serialization for the `NoMatch` component, alongside a number of context providers (`RouterContext`, `CurrentUrlContext`). If you wish to omit some features from your final bundle, you can skip the `Router` component and render those primitives instead.
+
+### `<Route />`
+
+Interacts with the `Router` instance to register behavior when the location matches a URL. There is only one mandatory prop, `match`, which specifies the URL to match. This prop can be one of the following:
+
+- a `string`, in which case it will match any URL with a pathname that exactly matches, or that exactly matches except for a trailing `/`,
+- a `RegExp`, in which case it will match any URL with a pathname that matches the regex, or,
+- a function that takes an [`EnhancedURL` object](#enhancedurl), and returns a boolean indicating whether or not to match.
+
+A `<Route />` can then specify the behavior to use for the matching URLs by specifying any set of the following props:
+
+- `render`: a function that is called with an `EnhancedURL` object, and returns JSX to use when the route matches.
```

Comment:
thoughts on `render` vs `children` for the primary render prop? Personally I like `children` better

Reply:
I find `children` to not be a great fit for this case because you will very likely also have `renderPrefetch` (at least for our case), and having them be in very different spots feels wrong to me.